### PR TITLE
Px4 support (WIP)

### PR DESCRIPTION
**Pull request by aftersomemath**
Looking for any initial feedback.

TODO:
* Test everything
* Move FC Comms to complete param loading
* Check that I can use raw_attitude commands the way I am
* Accompanying launch file PR

### CODE REVIEW

**Comment by aftersomemath** [comment]
PX4 does not support mixing command modes in offboard mode meaning we can't send orientation and yaw rate requests at the same time. So we will have to move to a full orientation system which will be interesting.

File: include/iarc7_fc_comms/CommonFcComms.hpp [hunk 1]
```diff
@@ -674,14 +686,14 @@ void CommonFcComms<T>::publishFcStatus()
 
 // Send out the orientation date from the quad
 template<class T>
-void CommonFcComms<T>::sendOrientation(double (&attitude)[3])
+void CommonFcComms<T>::sendOrientation(double (&attitude)[3], ros::Time attitude_stamp)
```
Comment (amiller27) [inline include/iarc7_fc_comms/CommonFcComms.hpp:689]: Can we pass this by `const&`?

File: include/iarc7_fc_comms/MspCommands.hpp [hunk 2]
```diff
@@ -192,7 +192,9 @@ namespace FcComms
 
             // Roll and pitch are in tenths of degrees, heading is in degrees
             attitude_values[0] = attitude_values[0] / 10.0 * M_PI / 180.0;
-            attitude_values[1] = attitude_values[1] / 10.0 * M_PI / 180.0;
+
+            // Invert pitch to match right handed coordinate system
+            attitude_values[1] = -1.0 * attitude_values[1] / 10.0 * M_PI / 180.0;
```
Comment (amiller27) [inline include/iarc7_fc_comms/MspCommands.hpp:197]: Did this sign change somewhere else?

File: include/iarc7_fc_comms/PX4FcComms.hpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+#include <mavros_msgs/AttitudeTarget.h>
+#include <mavros_msgs/CommandBool.h>
+#include <mavros_msgs/SetMode.h>
+#include <mavros_msgs/State.h>
+#include <sensor_msgs/Imu.h>
+
+namespace FcComms
+{
+    // Inherit from CommonFcComms to get the stuff any Fc is required to do.
```
Comment (amiller27) [inline include/iarc7_fc_comms/PX4FcComms.hpp:29]: Is this supposed to inherit from `CommonFcComms`?

File: include/iarc7_fc_comms/MspCommands.hpp (hunk 2 above)
Comment (aftersomemath) [reply include/iarc7_fc_comms/MspCommands.hpp:197]: Yes in CommonFcComms.....

**Comment by asaba96** [comment]
Is there any update for this?

**Comment by aftersomemath** [comment]
Unfortunately no

**Pull request closed without merging**